src/client/MenuWindow.cpp

```diff
@@ -2,186 +2,372 @@
 
-        for (unsigned index = 0; index < data.size(); ++index)
-        {
-            menuButtons.emplace_back(   sf::Vector2f(data[index]["width"].asFloat() * gameTitle->getLocalBounds().width, data[index]["height"].asFloat() * gameTitle->getLocalBounds().height), 
-                                        sf::Vector2f(   gameTitle->getPosition().x + data[index]["x"].asFloat() * gameTitle->getLocalBounds().width, 
-                                                        gameTitle->getPosition().y - data[index]["y"].asFloat() * gameTitle->getLocalBounds().height), buttonColor);
-            menuButtons.back().setText(40, sf::Vector2f(0, 0), data[index]["text"].asString(), *menuFont);
-        }
+    if (!menuFont.loadFromFile(RESOURCES_PATH "/img/hud/font.otf"))
+    {
+        std::cerr << "Font not loaded" << std::endl;
     }
 
-    /*!
-     * @brief Function that deteck where the user click and what to send to the engine
-     * @param timeSecond is a boolean used to 
-     */
-    long MenuWindow::getCurrentTime(bool timeSecond)
+    std::ifstream file(RESOURCES_PATH "/img/menu/menu.json");
+    if (!file.is_open())
     {
-        if (timeSecond)
-        {
-            return std::chrono::duration_cast<std::chrono::seconds>(std::chrono::system_clock::now().time_since_epoch()).count();
-        } 
-        else 
+        std::cerr << "Error while opening json ressources file /img/menu/menu.json" << std::endl;
+        exit(1);
+    }
+    std::string str((std::istreambuf_iterator<char>(file)), std::istreambuf_iterator<char>());
+
+    std::unique_ptr<Json::CharReader> reader = std::unique_ptr<Json::CharReader>(Json::CharReaderBuilder().newCharReader());
+    Json::Value obj;
+    std::string errors;
+    reader->parse(str.c_str(), str.c_str() + str.size(), &obj, &errors);
+
+    loadText(obj["welcomeTexts"]);
+    loadButton(obj["welcomeButtons"]);
+
+    currentMenu = &newGameButtons;
+    currentText = &newGameTexts;
+    
+    newGameButtons.push_back(menuButtons[BUTTON_USERNAME]);
+    newGameButtons.push_back(menuButtons[BUTTON_SERVER]);
+    newGameButtons.push_back(menuButtons[BUTTON_PORT]);
+    
+    loadText(obj["newGameTexts"]);
+    loadButton(obj["newGameButtons"]);
+
+    currentMenu = &menuButtons;
+    currentText = &menuTexts;
+}
+
+void MenuWindow::loadText(Json::Value &data)
+{
+    for (auto &dataMenu : data)
+    {
+        currentText->emplace_back(dataMenu["text"].asString(), menuFont, dataMenu["size"].asInt());
+        if (dataMenu["bold"].asBool())
         {
-            return std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::system_clock::now().time_since_epoch()).count();
+            currentText->back().setStyle(sf::Text::Bold);
         }
+        currentText->back().setFillColor(sf::Color::Black);
+        currentText->back().setPosition((int)(gameEnginePtr->clientWindow->getSize().x 
```

Comment:
You do these operation at least 4 times (to the nearest sign). This need a function